src/scripts/main.js

```diff
@@ -1,3 +1,416 @@
+      if (gameBoard[i][j] !== null) {
+        let k = j + 1;
+
+        while (k < gameBoard[i].length && (gameBoard[i][k] === null
+          || gameBoard[i][k] === gameBoard[i][j])) {
+          if (gameBoard[i][k] === gameBoard[i][j]) {
+            gameBoard[i][k] *= 2;
+
+            countScore(gameBoard[i][k]);
+
+            const cell = document.querySelector(`.game-field
+            tr:nth-child(${i + 1})
+            td:nth-child(${k + 1})`);
+
+            cell.classList.add('zoom-in');
+
+            cell.addEventListener('animationend', () => {
+              cell.classList.remove('zoom-in');
+            });
+
+            gameBoard[i][j] = null;
+
+            registerAction++;
+
+            break;
+          } else if (gameBoard[i][k] === null) {
+            k++;
+          }
+        }
+
+        if (k !== j + 1) {
+          gameBoard[i][k - 1] = gameBoard[i][j];
+          gameBoard[i][j] = null;
+
+          registerAction++;
+        }
+      }
+    }
+  }
+
+  if (registerAction !== 0) {
+    generateTiles();
+    registerAction = 0;
+  }
+}
+
+function isGameOver() {
+  for (let i = 0; i < gameBoard.length; i++) {
+    for (let j = 0; j < gameBoard[i].length; j++) {
+      if (gameBoard[i][j] === null) {
+        return false;
+      }
+
+      if (
+        (i > 0 && gameBoard[i][j] === gameBoard[i - 1][j])
+        || (i < gameBoard.length - 1
+          && gameBoard[i][j] === gameBoard[i + 1][j])
+        || (j > 0 && gameBoard[i][j] === gameBoard[i][j - 1])
+        || (j < gameBoard[i].length - 1
+          && gameBoard[i][j] === gameBoard[i][j + 1])
+      ) {
+        return false;
+      }
+    }
+  }
+
+  const infoLose = document.querySelector('.message-lose');
+
+  infoLose.classList.remove('hidden');
+}
+
+function isWin() {
+  for (let i = 0; i < gameBoard.length; i++) {
+    for (let j = 0; j < gameBoard[i].length; j++) {
+      if (gameBoard[i][j] === 2048) {
+        const infoWin = document.querySelector('.message-win');
+
+        infoWin.classList.remove('hidden');
+      }
+    }
+  }
+}
+
+document.addEventListener('keydown', keyEvent => {
+  const up = keyEvent.key === 'ArrowUp' || keyEvent.key === 'w';
+  const right = keyEvent.key === 'ArrowRight' || keyEvent.key === 'd';
+  const down = keyEvent.key === 'ArrowDown' || keyEvent.key === 's';
+  const left = keyEvent.key === 'ArrowLeft' || keyEvent.key === 'a';
+
+  if (gameStatus === 'on') {
+    if (up || right || down || left) {
+      keyEvent.preventDefault();
+    }
+
+    if (up) {
+      moveUp();
+      showBoardToUser(gameBoard);
+      isGameOver();
+      isWin();
+    } else if (right) {
+      moveRight();
+      showBoardToUser(gameBoard);
+      isGameOver();
+      isWin();
+    } else if (down) {
+      moveDown();
+      showBoardToUser(gameBoard);
+      isGameOver();
+      isWin();
+    } else if (left) {
+      moveLeft();
+      showBoardToUser(gameBoard);
+      isGameOver();
+      isWin();
+    }
+  }
```

Comment:
better to use switch case instead of multiple if else

and

let's move out from the switch case these three functions
```
 showBoardToUser(gameBoard);
      isGameOver();
      isWin();
```

Reply:
Ok done. Thanks for advice!